simplify phase, time, amp correlation code

ahnitz opened the pull request:
This simplifies the file format and the generation of the phase / time / correlation code. It also removes some of the manual settings related to number of bins etc, and replaces them with choices based on expected errors, etc. 

The format no longer stores ratio and magnitude, and instead stores the individual SNRs directly. This removes one of the failure modes for compensating for unequal detector sensitivity. It also makes it clearer how to do the smoothing of the distribution along this axis. 

Another axis for the ratio of detector sensitivity is instead added to simplify the interpretation. 

The code now can be run much faster and produce a viable file, in large part due to the introduction of the smoothing of values across the dimensions. 

The following runs in ~20 minutes on a head node and returns a reasonable file. (getting more points doesn't seem to affect recovery). 

'''
pycbc_stat_dtphase \
--ifos H1 L1 \
--sample-size 10000000 \
--min-snr 5.5 \
--max-snr 30 \
--timing-error .0005 \
--snr-error 1 \
--cores 14 \
--bin-density 3 \
--coinc-threshold .005 \
--output-file test.hdf \
--verbose
'''

tdent commented:
Could you put in some help messages for the histogram/smoothing options on the new executable, then should be OK to push

ahnitz commented:
@tdent I've updated the branch with some better help messages, and also added an approximation of the phase error that is derived from the snr error.
